core/src/test/java/org/bitcoinj/core/TransactionTest.java

```diff
@@ -669,100 +669,62 @@ static <T, R extends T> IntFunction<R[]> genericArray(IntFunction<T[]> arrayCrea
         return size -> (R[]) arrayCreator.apply(size);
     }
 
-    @Test
+    @Test(expected = BufferUnderflowException.class)
```

Comment:
I suppose using the `expected` feature to catch and validate the exception could be a separate commit.

Reply:
I've not yet done this second commit (actually 2 seconds commits). I hope it's understandable as is now. But if you think it's necessary, I can do it.

Reply:
I don't think it's necessary.